### PR TITLE
Fixes #15 - Remove use of base_url and update readme

### LINKED ISSUE #15: Remove use of base_url and update readme
We don't need a base_url for fxapom, so all references to it should be removed.
Also, the readme should be updated to remove/replace references to mozwebqa.

### PR DESCRIPTION
**Pull request by bobsilverberg**
@davehunt r?

### CODE REVIEW

File: fxapom/pages/sign_in.py [hunk 1]
```diff
@@ -38,32 +38,20 @@ def __init__(self, base_url, selenium, timeout=TIMEOUT):
...
         self.click_sign_in()
 
-    @property
-    def email(self):
-        """Get the value of the email field."""
-        return self.selenium.find_element(*self._email_locator).get_attribute('value')
-
-    @email.setter
-    def email(self, value):
+    def type_email(self, value):
```
Comment (davehunt) [inline fxapom/pages/sign_in.py]: Why are you making this change? Personally I prefer the setters, even though we don't use that approach much at the moment. This would break any consumers of FxAPOM that set these values directly (admittedly, there are probably none that do this).

File: fxapom/pages/sign_in.py [hunk 2]
```diff
@@ -38,32 +38,20 @@ def __init__(self, base_url, selenium, timeout=TIMEOUT):
...
         if self.is_element_visible(*self._next_button_locator):
             self.click_next()
-        self.login_password = password
+        self.type_password(password)
         self.click_sign_in()
 
-    @property
-    def email(self):
-        """Get the value of the email field."""
-        return self.selenium.find_element(*self._email_locator).get_attribute('value')
```
Comment (davehunt) [inline fxapom/pages/sign_in.py]: @silne30 noticed that the locator here is incorrect and should be `self._email_input_locator`.

**Comment by davehunt** [comment]
r+ with the reintroduction of the getters & setters, but I'm open to discussion on this.

**Comment by bobsilverberg** [comment]
The change that involves removing the getters and setters started with me noticing that the getters referenced a non-existent locator (or, as @silne30 suggests perhaps just a misspelled locator). I was surprised by that and then checked to see if these getters are used anywhere and found that they are not, so I felt they should be removed. Simply removing them causes a problem with the setters as the setters expect the getter property to exist.

Contrary to your preference, I actually don't like the setters as they are inconsistent with pretty much every other piece of code we've written. While I don't oppose them in principle, I don't like using a construct that is not used anywhere else in our code and I believe could be quite confusing to users because of that. In a test that uses the PO you'd need to know that the setter exists because the syntax for calling it is very different than calling a method that does something. This is what I think could be quite confusing to users who have experience with any of our other projects that use page objects.

To sum up:
1. I don't like having properties in an object that are not used. I believe we tend to follow a model where we don't add properties and methods to page objects unless they are needed by a test. We don't add things that we think *might* be needed in the future.
2. If I remove the getter property, I believe it becomes impossible (ok, probably not impossible, but would need more code) to create the setter for the property.
3. While I do like property getters and setters in principle, I don't like the inconsistency of using setters here but nowhere else.

@davehunt What do you think?

**Comment by davehunt** [comment]
I'll respond to each of your points:

> I don't like having properties in an object that are not used. I believe we tend to follow a model where we don't add properties and methods to page objects unless they are needed by a test. We don't add things that we think might be needed in the future.

These are public APIs so we can't say they're not used. I agree that they should have some sort of test within the project, which would have prevented the broken locator references. The property most likely primarily exists to allow us to create a setter.

> If I remove the getter property, I believe it becomes impossible (ok, probably not impossible, but would need more code) to create the setter for the property.

No idea if it's impossible, but I wouldn't spend much time trying to find out if it is.

> While I do like property getters and setters in principle, I don't like the inconsistency of using setters here but nowhere else.

We use it in BIDPOM, which was probably the model when creating FxAPOM. Personally I would prefer page objects to offer getters and setters instead of the various `type_`, `select_`, methods. Just because we don't currently recommend this approach it doesn't mean we can't start doing that - I believe that was the reason we did it in the first place with BIDPOM.

Let's remove this change from this pull request as it's unrelated to the base_url change. We can always open another pull request or issue to discuss this further.

**Comment by bobsilverberg** [comment]
I do like the ideas of getters and setters, and considering that I don't think that anyone is using any of the methods currently anyway I guess it's fine to try to start a new standard. I can remove the change, but I will at least need to update the locators in the getters. I don't want to keep broken code in there.

**Comment by bobsilverberg** [comment]
@davehunt ok, it's ready for another look from you

**Comment by davehunt** [comment]
r+

**Comment by bobsilverberg** [comment]
Thanks @davehunt. I also released version 1.6 to PyPI.
